### PR TITLE
Fix relative canonical link (See #228)

### PR DESCRIPTION
**Pull request by adamn**
PR #228 looks good but it wasn't handling a meta url like 'example.com/article.html', only urls like 'article.html' would have been parsed correctly.

I added a new test and refactored `get_canonical_link()` accordingly

### CODE REVIEW

**Comment by yprez** [comment]
Looks good to me.
@codelucas can you take a look as well?

**Comment by codelucas** [comment]
Great catch! Also thanks so much for also adding unit tests 👍 
Merged.

File: newspaper/extractors.py [hunk 1]
```diff
@@ -511,26 +511,35 @@ def get_canonical_link(self, article_url, doc):
...
+                            format(parsed_article_url.hostname),
+                                  parsed_meta_url.path)
+                try:
+                    true_path = strip_hostname_in_meta_path.group(1)
+                except AttributeError:
+                    true_path = parsed_meta_url.path
+
+                # true_path may contain querystrings and fragments
+                meta_url = urlunparse((parsed_article_url.scheme,
+                                       parsed_article_url.hostname,true_path,
```
Comment (codelucas) [inline newspaper/extractors.py]: I would have put true_path on its own line, but thanks for fixing my python style in this 👍

Comment (adamn) [inline newspaper/extractors.py]: Yeah that was me just using cmd+shift+l in PyCharm.

**Pull request merged**